Append playbook to uninstall metrics #2726

harobed opened the pull request:
Append playbook to uninstall metrics

Fixes #2726

rhcarvalho commented:
@harobed thanks for the PR! Could you please update the PR description to include "fixes #\<issue-number\>"?

https://github.com/blog/1506-closing-issues-via-pull-requests

This way, whenever your PR gets merged the issue will be closed automatically.

harobed commented:
@rhcarvalho Could you please update the PR description to include "fixes #<issue-number>"?

Done :)

rhcarvalho commented:
The description I see here still says "Append playbook to uninstall metrics #2726", it won't close the issue automatically. Could you please check it again?

rhcarvalho commented on playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,21 @@
+    command: "{{item}}"
+    with_items:
+      - oc delete all --selector=metrics-infra --namespace openshift-infra --ignore-not-found
+      - oc delete templates --selector=metrics-infra --namespace openshift-infra --ignore-not-found
```
tip: you can pass multiple resource types separated by comma:

```
oc delete all,templates --selector=metrics-infra --namespace openshift-infra --ignore-not-found
```

rhcarvalho commented on playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,21 @@
+      - oc delete secret hawkular-cassandra-secrets --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-account --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-certificate --namespace openshift-infra --ignore-not-found
+      - oc delete secret hawkular-metrics-secrets --namespace openshift-infra --ignore-not-found
```
I'm not familiar with these metrics resources, isn't there a label to select them all? If so, I'd suggest a single call to `oc delete`.

```
oc -n openshift-infra delete --ignore-not-found all,templates,sa,secret -l metrics-infra
```

rhcarvalho commented:
> The description I see here still says "Append playbook to uninstall metrics #2726", it won't close the issue automatically. Could you please check it again?

Aha, I see what happened. The commit message was updated, but not the PR description. I've updated it now.

rhcarvalho commented:
aos-ci-test

harobed commented:
> @harobed thanks for the PR! Could you please update the PR description to include "fixes #<issue-number>"?

Fixed

> I have no experience with the metrics setup, but wanted to share some tips wrt oc delete usage.

Fixed

rhcarvalho commented on playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,11 @@
+- hosts: oo_first_master
+  tasks:
+  - name: "Remove metrics url from  master config"
+    lineinfile: "state=absent dest=/etc/origin/master/master-config.yaml regexp='^\ \ metricsPublicURL'"
```
https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#Parameters-to-Ansible-modules-SHOULD-use-the-Yaml-dictionary-format-when-3-or-more-parameters-are-being-passed

rhcarvalho commented on playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,11 @@
+  - name: "Delete metrics objects"
+    command: "{{item}}"
+    with_items:
+      - oc delete all,secrets,templates,sa -l metrics-infra --namespace openshift-infra --ignore-not-found
```
With this oneliner I believe we don't need to loop over items?

rhcarvalho replied:
Please replace `oc` with `{{ openshift.common.client_binary }}`

rhcarvalho commented on playbooks/common/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,11 @@
+    command: "{{item}}"
+    with_items:
+      - oc delete all,secrets,templates,sa -l metrics-infra --namespace openshift-infra --ignore-not-found
+        
```
This seems to be missing some things, like removal of role bindings:

- [roles/openshift_metrics/tasks/install.yml#L31-L38](https://github.com/openshift/openshift-ansible/blob/bf0f8266e91b552ffa815d3a7bcb65f105606438/roles/openshift_metrics/tasks/install.yml#L31-L38)
- [roles/openshift_metrics/tasks/install.yml#L49-L56](https://github.com/openshift/openshift-ansible/blob/bf0f8266e91b552ffa815d3a7bcb65f105606438/roles/openshift_metrics/tasks/install.yml#L49-L56)

rhcarvalho commented on playbooks/byo/openshift-cluster/metrics/uninstall_metrics.yml:
```diff
@@ -0,0 +1,20 @@
+- hosts: localhost
```
Most yml files in the repo begin with `---`, could you please add that for consistency?

rhcarvalho commented:
@jcantrill FYI, I think the new metrics playbooks include uninstall?

rhcarvalho commented:
Here're the new metrics playbooks: https://github.com/openshift/openshift-ansible/pull/2873

@jcantrill, is this PR useful for existing installations?

mtnbikenc commented:
Closing this PR as it is stale.  Please reopen if you feel this patch is still required.
